wrappers/services.go

```diff
@@ -204,12 +204,56 @@ type StartServicesFlags struct {
 // are services. Service units will be started in the order provided by the
 // caller.
 func StartServices(apps []*snap.AppInfo, disabledSvcs []string, flags *StartServicesFlags, inter interacter, tm timings.Measurer) (err error) {
+	if flags == nil {
+		flags = &StartServicesFlags{}
+	}
 	systemSysd := systemd.New(dirs.GlobalRootDir, systemd.SystemMode, inter)
 	userSysd := systemd.New(dirs.GlobalRootDir, systemd.GlobalUserMode, inter)
 	cli := client.New()
 
+	var disableEnabledServices func()
+
+	defer func() {
+		if err == nil {
+			return
+		}
+		if disableEnabledServices != nil {
+			disableEnabledServices()
+		}
+	}()
+
+	var toEnable []*snap.AppInfo
 	systemServices := make([]string, 0, len(apps))
 	userServices := make([]string, 0, len(apps))
+
+	for _, app := range apps {
```

Comment:
this loop could explain that this loop gathers all non-sockets and non-timers together to enable first before starting

Reply:
Done.